### PR TITLE
construct sumcheck circuit

### PR DESCRIPTION
**Pull request by Michael-EY**
Constructed circuit to perform sumcheck verification. Also constructed a circuit to perform the evaluation of an oracle to a polynomial.

### CODE REVIEW

File: plonk/src/nightfall/circuit/subroutine_verifiers/sumcheck.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
+            assert_eq!(circuit.witness(result_var)?, evaluation);
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_sum_check_circuit() {
+        test_sum_check_circuit_template::<SWGrumpkin, _, _>().expect("test failed for bls12-381");
+    }
```
Comment (Zyouell) [inline plonk/src/nightfall/circuit/subroutine_verifiers/sumcheck.rs:121]: This is super pedantic I admit but I'm going to say it anyway the error here should say "test failed for Grumpkin".

File: plonk/src/nightfall/circuit/subroutine_verifiers/mod.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//! This module provides the circuits for the subroutines within Plonk verifiers.
+
+mod poly;
+mod structs;
+mod sumcheck;
```
Comment (Zyouell) [inline plonk/src/nightfall/circuit/subroutine_verifiers/mod.rs:5]: So in my opinion a nice way to wrap all of the functions you've written together would be to define a trait called `SumCheckGadget<F: PrimeField>` with methods like `evaluate_poly_oracle_var()`, `create_poly_oracle_var()`, `create_sumcheck_proof_var()` and `verify_sumcheck()`. Then you would implement this trait on the `PlonkCircuit<F>` struct and in the implementation for each method just call the relevant function that you have already written. 

A good example of this is in the `relation/src/gadgets/ecc/msm.rs` file where the `MultilScalarMultiplicationCircuit` trait is defined on line 20 and then implemented on a `PlonkCircuit<F>` on line 45 but this implementation essentially just wraps some stand alone functions written elsewhere in the code base.

**Review by Zyouell — COMMENTED** [review]
Looks very good to me just a couple of comments to make the code a bit easier to handle when we want to use it, mainly so that we can use `circuit.verify_sumcheck()`.

**Comment by Michael-EY** [comment]
Updated to address Zack's comments

**Review by Zyouell — APPROVED** [review]

**Pull request merged**